### ISSUE TITLE
image file name in ImageRecordIter

### ISSUE THREAD

**Issue by albertyou2** [dmlc/mxnet#4115]
Hello , Is there any method to get a image file name with ImageRecordIter?
My case is to extract features of each image with pre-trained model.Then I will do some image clustering with these features. 
Now I can extract features by using these code :
...
data_shape = (args.rgb, args.shape,args.shape)    
images = mx.io.ImageRecordIter(  
        path_imgrec = args.data_file,
        mean_img    = args.min_file,  
        rand_crop    = False,  
        rand_mirror = False,  
        data_shape = data_shape,  
        batch_size   = 1) 
model = mx.model.FeedForward.load(args.prefix, args.iter)
internals = model.symbol.get_internals()
layer = args.layer+'_output'
fea_symbol = internals[layer]
feature_extractor = mx.model.FeedForward(ctx=mx.cpu(), symbol=fea_symbol, numpy_batch_size=1,
                                         arg_params=model.arg_params, aux_params=model.aux_params,
                                         allow_extra_params=True)
global_pooling_feature = feature_extractor.predict(images)
a = np.asarray(global_pooling_feature)
np.savetxt('./pred_'+args.layer+'.csv',a)
...
But I don't know each feature belongs to which image.
I want to create a .csv file with this format : filename , f1 , f2 , ..... , fn
is there a way to do this ?
should I set the "shuffle     = True" in ImageRecordIter first?

**Comment by piiswrong** [comment]
if shuffle=false the order corresponds to the order in your .lst file that's used to generate the .rec
and .lst has filename in it

**Comment by albertyou2** [comment]
ok ,thank you  
i got it
